<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Anna Hutchinson" />


<title>Useful Info</title>






<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Useful Info</h1>
<h4 class="author">Anna Hutchinson</h4>



<p>This vignette contains supplementary information regarding the usage of the <code>corrcoverage</code> R package.</p>
<hr />
<div id="key-functions" class="section level2">
<h2>Key Functions</h2>
<p>The two main functions are:</p>
<ol style="list-style-type: decimal">
<li><p><code>corrcov</code> (or analogously <code>corrcov_bhat</code>): Provides a corrected coverage estimate of credible sets obtained using the Bayesian approach for fine-mapping (see ‘Corrected Coverage’ vignette). Use <code>corrcov</code> for <span class="math inline">\(Z\)</span>-scores and MAFs or <code>corrcov_bhat</code> for beta hat estimates and their standard errors.</p></li>
<li><p><code>corrected_cs</code> (or analogously <code>corrected_cs_bhat</code>): Finds a new ‘corrected’ credible set, which is the smallest set of variants such that the corrected coverage is above some user defined ‘desired coverage’. Use <code>corrected_cs</code> for <span class="math inline">\(Z\)</span>-scores and MAFs or <code>corrected_cs_bhat</code> for beta hat estimates and their standard errors.</p></li>
</ol>
<hr />
</div>
<div id="other-useful-functions" class="section level2">
<h2>Other Useful Functions</h2>
<ul>
<li><p><code>corrcov_nvar</code> (or analogously <code>corrcov_nvar_bhat</code>): Finds a corrected coverage estimate whereby the simulated credible sets used to derive the estimate are limited to those which contain a specified number of variants (parameter ‘nvar’). These functions should be used with caution and only ever for very small credible sets (fewer than 4 variants).</p></li>
<li><p><code>corrcov_CI</code> (or analogously <code>corrcov_CI_bhat</code>): Finds a ‘confidence interval’ for the corrected coverage estimate. The default is a 95% confidence interval (parameter ‘CI’ = 0.95), but the ‘CI’ parameter can be adjusted accordingly. This function involves repeating the correction procedure 100 times and so, depending on the number of SNPs, may take a long time to implement.</p></li>
</ul>
<hr />
</div>
<div id="conversion-functions" class="section level2">
<h2>Conversion Functions</h2>
<p>The Bayesian method for fine-mapping involves finding the posterior probability of causality for each SNP, before sorting these into descending order and adding variants to a ‘credible set’ until the combined posterior probabilities of these SNPs exceed some threshold. The supplementary text of Maller’s paper (available <a href="https://media.nature.com/original/nature-assets/ng/journal/v44/n12/extref/ng.2435-S1.pdf">here</a>) shows that these posterior probabilities are normalised Bayes factors.</p>
<p>Asymptotic Bayes factors (<a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/gepi.20359">Wakefield, 2009</a>) are commonly used in genetic association studies as these only require the specification of <span class="math inline">\(Z\)</span>-scores (or equivalently the effect size coefficients, <span class="math inline">\(\beta\)</span>, and their standard errors, <span class="math inline">\(V\)</span>), the standard errors of the effect sizes (<span class="math inline">\(V\)</span>) and the prior variance of the estimated effect size (<span class="math inline">\(W^2\)</span>), thus only requiring summary data from genetic association studies plus an estimate for the <span class="math inline">\(W\)</span> parameter.</p>
<p>Consequently, the <code>corrcoverage</code> package contains functions for converting between <span class="math inline">\(P\)</span>-values, <span class="math inline">\(Z\)</span>-scores, asymptotic Bayes factors (ABFs) and posterior probabilities of causality (PPs). The following table shows what input these conversion functions require and what output they produce. The ‘include null model’ column is for whether the null model of no genetic effect is included in the calculation (PPs obtained using the standard Bayesian approach ignore this).</p>
<table>
<thead>
<tr class="header">
<th>Function</th>
<th>Include null model?</th>
<th>Input</th>
<th>Output</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>approx.bf.p</code></td>
<td>YES</td>
<td><span class="math inline">\(P\)</span>-values</td>
<td>log(ABF)</td>
</tr>
<tr class="even">
<td><code>pvals_pp</code></td>
<td>YES</td>
<td><span class="math inline">\(P\)</span>-values</td>
<td>Posterior Probabilities</td>
</tr>
<tr class="odd">
<td><code>z0_pp</code></td>
<td>YES</td>
<td>Marginal <span class="math inline">\(Z\)</span>-scores</td>
<td>Posterior Probabilities</td>
</tr>
<tr class="even">
<td><code>ppfunc</code></td>
<td>NO</td>
<td>Marginal <span class="math inline">\(Z\)</span>-scores</td>
<td>Posterior Probabilities</td>
</tr>
</tbody>
</table>
<p>Functions are also provided to simulate marginal <span class="math inline">\(Z\)</span>-scores from joint <span class="math inline">\(Z\)</span>-scoresm(<span class="math inline">\(Z_j\)</span>). The joint <span class="math inline">\(Z\)</span>-scores are all 0, except at the causal variant where it is the ‘true effect’, <span class="math inline">\(\mu\)</span>.</p>
<p><span class="math inline">\(\mu\)</span> can be estimated using the <code>est_mu</code> function which required sample sizes, <span class="math inline">\(Z\)</span>-scores and minor allele frequencies.</p>
<p><code>z_sim</code> simulates marginal <span class="math inline">\(Z\)</span>-scores from joint <span class="math inline">\(Z\)</span>-scores, whilst <code>zj_pp</code> can be used to simulate posterior probability systems from a joint <span class="math inline">\(Z\)</span>-score vector. These functions first calculate the expected marginal <span class="math inline">\(Z\)</span> scores, <span class="math inline">\(E(Z)\)</span>, <span class="math display">\[E(Z)=Z_j \times \Sigma\]</span> where <span class="math inline">\(\Sigma\)</span> is the correlation matrix between SNPs.</p>
<p>We can then simulate more <span class="math inline">\(Z\)</span>-score systems from a multivariate normal distribution with mean <span class="math inline">\(E(Z)\)</span> and variance <span class="math inline">\(\Sigma\)</span>. This is a key step in our corrected coverage method.</p>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
