<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Correcting the Coverage of Credible Sets • corrcoverage</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Correcting the Coverage of Credible Sets">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">corrcoverage</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/my-vignette.html">Correcting the Coverage of Credible Sets</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Correcting the Coverage of Credible Sets</h1>
                        <h4 class="author">Anna Hutchinson</h4>
            
            <h4 class="date">2019-02-06</h4>
      
      
      <div class="hidden name"><code>my-vignette.Rmd</code></div>

    </div>

    
    
<hr>
<p>This guide will show readers how to use the <code>corrcoverage</code> package for fine-mapping analyses, including obtaining an accurate coverage estimate for the causal variant in a credible set. This package is specific to credible sets obtained using the Bayesian approach to fine-mapping, described by Maller et al. <a href="https://www.ncbi.nlm.nih.gov/pubmed/23104008">here</a>.</p>
<p><code>corrcoverage</code> takes GWAS summary statistics (minor allele frequencies, z-scores (or p-values), sample sizes, reference LD matrix) and provides an accurate estimate of the probability that the true causal variant is contained within a credible set derived for a given threshold.</p>
<hr>
<p>Load the library.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">2</span>)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(corrcoverage)</code></pre></div>
<hr>
<div id="simulate-gwas-summary-statistics" class="section level2">
<h2 class="hasAnchor">
<a href="#simulate-gwas-summary-statistics" class="anchor"></a>Simulate GWAS summary statistics</h2>
<hr>
<p><strong>This package is intended for use on real summary statistics obtained from GWAS, such as observed z-scores. For the purpose of this vignette, we will simulate artificial haplotypes and GWAS data.</strong></p>
<p>We use the <a href="https://github.com/chr1swallace/simGWAS">simGWAS</a> package to simulate GWAS data. Please refer to the walkthrough guide <a href="https://chr1swallace.github.io/simGWAS/articles/intro.html">here</a>, from which the following is based. Start by loading this package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(simGWAS)</code></pre></div>
<p>Next, simulate some reference haplotypes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nsnps &lt;-<span class="st"> </span><span class="dv">200</span>
nhaps &lt;-<span class="st"> </span><span class="dv">1000</span>
lag &lt;-<span class="st"> </span><span class="dv">5</span> <span class="co"># genotypes are correlated between neighbouring variants</span>
maf &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/Uniform">runif</a></span>(nsnps<span class="op">+</span>lag,<span class="fl">0.05</span>,<span class="fl">0.5</span>) <span class="co"># common SNPs</span>
laghaps &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/do.call">do.call</a></span>(<span class="st">"cbind"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">lapply</a></span>(maf, <span class="cf">function</span>(f) <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/Binomial">rbinom</a></span>(nhaps,<span class="dv">1</span>,f)))
haps &lt;-<span class="st"> </span>laghaps[,<span class="dv">1</span><span class="op">:</span>nsnps]
<span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>lag) 
    haps &lt;-<span class="st"> </span>haps <span class="op">+</span><span class="st"> </span>laghaps[,(<span class="dv">1</span><span class="op">:</span>nsnps)<span class="op">+</span>j]
haps &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Round">round</a></span>(haps<span class="op">/</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">matrix</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/apply">apply</a></span>(haps,<span class="dv">2</span>,max),nhaps,nsnps,<span class="dt">byrow=</span><span class="ot">TRUE</span>))
snps &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">colnames</a></span>(haps) &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(<span class="st">"s"</span>,<span class="dv">1</span><span class="op">:</span>nsnps)
freq &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>(haps<span class="op">+</span><span class="dv">1</span>)
freq<span class="op">$</span>Probability &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">/</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/nrow">nrow</a></span>(freq)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sum">sum</a></span>(freq<span class="op">$</span>Probability)</code></pre></div>
<pre><code>## [1] 1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">MAF &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colSums">colMeans</a></span>(freq[,snps]<span class="op">-</span><span class="dv">1</span>)</code></pre></div>
<p>Next, we need to specify the causal variant and its effect on disease, as odds ratio. We only choose one CV as this is a key assumption underlying the Bayesian approach for fine-mapping. We choose a CV from the column names in <code>freq</code> and a value for the odds ratio.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">CV &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sample">sample</a></span>(snps[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colSums">colMeans</a></span>(haps)<span class="op">&gt;</span><span class="fl">0.1</span>)],<span class="dv">1</span>)
iCV &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">sub</a></span>(<span class="st">"s"</span>,<span class="st">""</span>,CV) <span class="co"># index of cv</span>
g1 &lt;-<span class="st"> </span><span class="fl">1.1</span></code></pre></div>
<p>We then simulate marginal z-scores. Here, we consider a GWAS with 5000 cases and 5000 controls.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">z0 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/simGWAS/topics/simulated_z_score">simulated_z_score</a></span>(<span class="dt">N0=</span><span class="dv">5000</span>, <span class="co"># number of controls</span>
              <span class="dt">N1=</span><span class="dv">5000</span>, <span class="co"># number of cases</span>
              <span class="dt">snps=</span>snps, <span class="co"># column names in freq of SNPs for which Z scores should be generated</span>
              <span class="dt">W=</span>CV, <span class="co"># causal variants, subset of snps</span>
              <span class="dt">gamma.W=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Log">log</a></span>(g1), <span class="co"># log odds ratios</span>
              <span class="dt">freq=</span>freq <span class="co"># reference haplotypes</span>
              )</code></pre></div>
<hr>
</div>
<div id="obtain-credible-set" class="section level2">
<h2 class="hasAnchor">
<a href="#obtain-credible-set" class="anchor"></a>Obtain credible set</h2>
<hr>
<p>We now use the <code>credsets</code> function in the <code>corrcoverage</code> package to obtain a credible set using the Bayesian approach for fine-mapping. In brief, the function follows these steps:</p>
<ol style="list-style-type: decimal">
<li>Order variants in descending order of posterior probabilities.</li>
<li>Cumulatively sum these variants until the specified threshold is exceeded.</li>
<li>Group these variants, forming the credible set.</li>
</ol>
<p>The function requires posterior probabilities as input and so we use the <code>ppfunc</code> function to convert the marginal z-scores to posterior probabilties. This function requires a value for <span class="math inline">\(V\)</span>, the variance of the estimate effect size, which can be easily calculated using the <code>Var.data.cc</code> function in the <code>coloc</code> package. The prior standard deviation of the effect size, <span class="math inline">\(W\)</span>, is an optional parameter with a default value of 0.2</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">N0 &lt;-<span class="st"> </span><span class="dv">5000</span> <span class="co"># number of controls</span>
N1 &lt;-<span class="st"> </span><span class="dv">5000</span> <span class="co"># number of cases</span>

var &lt;-<span class="st"> </span>coloc<span class="op">:::</span><span class="kw"><a href="../reference/Var.data.cc.html">Var.data.cc</a></span>(<span class="dt">f =</span> MAF, <span class="dt">N =</span> N1<span class="op">+</span>N0, <span class="dt">s =</span> N1<span class="op">/</span>(N0<span class="op">+</span>N1)) <span class="co"># variance of estimated effect size</span>

postprobs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/ppfunc.html">ppfunc</a></span>(<span class="dt">z =</span> z0, <span class="dt">V =</span> var)

<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/plot">plot</a></span>(postprobs, <span class="dt">main =</span> <span class="st">"Posterior probabilities of variants"</span>)
<span class="kw"><a href="https://www.rdocumentation.org/packages/graphics/topics/abline">abline</a></span>(<span class="dt">v =</span> iCV, <span class="dt">col =</span> <span class="dv">2</span>)</code></pre></div>
<p><img src="my-vignette_files/figure-html/unnamed-chunk-6-1.png" width="384"></p>
<p>The posterior probability plot shows the location of the causal variant (red line) - we see that the causal variant is not the variant with the largest posterior probability. This is sometimes seen in fine-mapping experiments due to linkage disequilibrium (correlation between SNPs).</p>
<p>The <code>credset</code> function reports the claimed coverage <a href="#fn1" class="footnoteRef" id="fnref1"><sup>1</sup></a> (the cumulative sum of the posterior probabilities of the variants in the set) and the number of variants in the credible set. If the <code>CV</code> parameter is supplied in the <code>credset</code> function, then the output also includes a binary indicator of whether the causal variant was contained in the credible set.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">thresh &lt;-<span class="st"> </span><span class="fl">0.9</span>

<span class="kw"><a href="../reference/credset.html">credset</a></span>(<span class="dt">pp =</span> postprobs,
      <span class="co"># CV = iCV,</span>
       <span class="dt">thr =</span> thresh 
       ) </code></pre></div>
<pre><code>##     claimed.cov nvar
## s30   0.9008649  114</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/credset.html">credset</a></span>(<span class="dt">pp =</span> postprobs,
              <span class="dt">CV =</span> iCV,
             <span class="dt">thr =</span> thresh 
       ) 

cs</code></pre></div>
<pre><code>##     claimed.cov covered nvar
## s30   0.9008649       1  114</code></pre>
<p>The credible set obtained contains 114 (out of 200) variants and the claimed coverage is 0.9004596. In papers, we would expect to see author report that they have found a 90% credible set of 45 variants, which they are 90% confident contains the true causal variant.</p>
<hr>
</div>
<div id="obtain-corrected-coverage-estimate" class="section level2">
<h2 class="hasAnchor">
<a href="#obtain-corrected-coverage-estimate" class="anchor"></a>Obtain corrected coverage estimate</h2>
<hr>
<p>Suppose we are suspicious of this coverage estimate and wish to find a more accurate coverage probability using this fancy new package. The only parameters we need to supply to the <code>corrcov</code> function are z0 (marginal z-scores), f (minor allele frequencies), N0 (number of controls), N1 (number of cases), Sigma (SNP correlation matrix) and the threshold of the credible set.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">LD &lt;-<span class="st"> </span><span class="kw"><a href="../reference/cor2.html">cor2</a></span>(haps) <span class="co"># correlation between SNPs</span>

corrected_cov_estimate &lt;-<span class="st"> </span><span class="kw"><a href="../reference/corrcov.html">corrcov</a></span>(z0, <span class="dt">f =</span> MAF, N0, N1, <span class="dt">Sigma =</span> LD, <span class="dt">thr =</span> thresh)</code></pre></div>
<pre><code>##   Corrected.Coverage Claimed.Coverage Threshold
## 1          0.9593843        0.9008649       0.9</code></pre>
<p>In this example, the 90% credible set may in actual fact have nearer to 96% coverage of the causal variant in the credible set. To see the accuracy of this estimate we simulate more credible sets from the same system and find what proportion of these contains the true causal variant.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">z0.tmp &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/simGWAS/topics/simulated_z_score">simulated_z_score</a></span>(<span class="dt">N0=</span><span class="dv">5000</span>, <span class="co"># number of controls</span>
                            <span class="dt">N1=</span><span class="dv">5000</span>, <span class="co"># number of cases</span>
                            <span class="dt">snps=</span>snps, <span class="co"># column names in freq</span>
                            <span class="dt">W=</span>CV, <span class="co"># causal variants, subset of snps</span>
                            <span class="dt">gamma.W=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Log">log</a></span>(g1), <span class="co"># log odds ratios</span>
                            <span class="dt">freq=</span>freq, <span class="co"># reference haplotypes</span>
                            <span class="dt">nrep =</span> <span class="dv">5000</span>
)

pps &lt;-<span class="st"> </span><span class="kw"><a href="../reference/ppfunc.mat.html">ppfunc.mat</a></span>(<span class="dt">zstar =</span> z0.tmp, <span class="dt">V =</span> var) <span class="co"># find pps</span>
cs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/apply">apply</a></span>(pps, <span class="dv">1</span>, credset, <span class="dt">CV =</span> iCV, <span class="dt">thr =</span> thresh) <span class="op">%&gt;%</span><span class="st"> </span>data.table<span class="op">:::</span><span class="kw">rbindlist</span>() <span class="co"># form cred sets</span>
true.cov.est &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/mean">mean</a></span>(cs<span class="op">$</span>covered) <span class="co"># average covered col</span>
true.cov.est</code></pre></div>
<pre><code>## [1] 0.9746</code></pre>
<p>The empirical coverage is found to be approximately 97.5%, so our estimate of ~96% is much more accurate than the original 90% (or 90.09% is using claimed coverage).</p>
<pre><code>##   Empirical.Coverage Corrected.Coverage Claimed.Coverage Threshold
## 1             0.9746          0.9593843        0.9005427       0.9</code></pre>
<hr>
</div>
<div id="references" class="section level2">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<ol style="list-style-type: decimal">
<li>
<a href="https://github.com/chr1swallace/simGWAS">simGWAS</a>: Mary D Fortune, Chris Wallace; simGWAS: a fast method for simulation of large scale case–control GWAS summary statistics, Bioinformatics, , bty898, <a href="https://doi.org/10.1093/bioinformatics/bty898" class="uri">https://doi.org/10.1093/bioinformatics/bty898</a>
</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">session_info</a></span>()</code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value                       
##  version  R version 3.5.1 (2018-07-02)
##  os       macOS High Sierra 10.13.6   
##  system   x86_64, darwin15.6.0        
##  ui       X11                         
##  language (EN)                        
##  collate  en_GB.UTF-8                 
##  ctype    en_GB.UTF-8                 
##  tz       Europe/London               
##  date     2019-02-06                  
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package      * version    date       lib source                               
##  assertthat     0.2.0      2017-04-11 [1] CRAN (R 3.5.0)                       
##  backports      1.1.3      2018-12-14 [1] CRAN (R 3.5.0)                       
##  bindr          0.1.1      2018-03-13 [1] CRAN (R 3.5.0)                       
##  bindrcpp       0.2.2      2018-03-29 [1] CRAN (R 3.5.0)                       
##  BiocGenerics   0.28.0     2018-10-30 [1] Bioconductor                         
##  BMA            3.18.9     2018-09-14 [1] CRAN (R 3.5.0)                       
##  callr          3.1.1      2018-12-21 [1] CRAN (R 3.5.0)                       
##  cli            1.0.1      2018-09-25 [1] CRAN (R 3.5.0)                       
##  cluster        2.0.7-1    2018-04-13 [1] CRAN (R 3.5.1)                       
##  codetools      0.2-15     2016-10-05 [1] CRAN (R 3.5.1)                       
##  coloc          3.1        2018-02-24 [1] CRAN (R 3.5.0)                       
##  colorspace     1.4-0      2019-01-13 [1] CRAN (R 3.5.2)                       
##  combinat       0.0-8      2012-10-29 [1] CRAN (R 3.5.0)                       
##  commonmark     1.7        2018-12-01 [1] CRAN (R 3.5.0)                       
##  corpcor        1.6.9      2017-04-01 [1] CRAN (R 3.5.0)                       
##  corrcoverage * 0.1.0      2019-02-06 [1] local                                
##  crayon         1.3.4      2017-09-16 [1] CRAN (R 3.5.0)                       
##  data.table     1.12.0     2019-01-13 [1] CRAN (R 3.5.2)                       
##  DEoptimR       1.0-8      2016-11-19 [1] CRAN (R 3.5.0)                       
##  desc           1.2.0      2018-05-01 [1] CRAN (R 3.5.0)                       
##  devtools       2.0.1      2018-10-26 [1] CRAN (R 3.5.1)                       
##  digest         0.6.18     2018-10-10 [1] CRAN (R 3.5.0)                       
##  dplyr          0.7.8      2018-11-10 [1] CRAN (R 3.5.0)                       
##  evaluate       0.12       2018-10-09 [1] CRAN (R 3.5.0)                       
##  fs             1.2.6      2018-08-23 [1] CRAN (R 3.5.0)                       
##  ggplot2        3.1.0      2018-10-25 [1] CRAN (R 3.5.0)                       
##  glue           1.3.0      2018-07-17 [1] CRAN (R 3.5.0)                       
##  gtable         0.2.0      2016-02-26 [1] CRAN (R 3.5.0)                       
##  htmltools      0.3.6      2017-04-28 [1] CRAN (R 3.5.0)                       
##  inline         0.3.15     2018-05-18 [1] CRAN (R 3.5.0)                       
##  knitr          1.21       2018-12-10 [1] CRAN (R 3.5.1)                       
##  lattice        0.20-35    2017-03-25 [1] CRAN (R 3.5.1)                       
##  lazyeval       0.2.1      2017-10-29 [1] CRAN (R 3.5.0)                       
##  leaps          3.0        2017-01-10 [1] CRAN (R 3.5.0)                       
##  magrittr       1.5        2014-11-22 [1] CRAN (R 3.5.0)                       
##  MASS           7.3-50     2018-04-30 [1] CRAN (R 3.5.1)                       
##  Matrix         1.2-14     2018-04-13 [1] CRAN (R 3.5.1)                       
##  memoise        1.1.0      2017-04-21 [1] CRAN (R 3.5.0)                       
##  mgcv           1.8-26     2018-11-21 [1] CRAN (R 3.5.0)                       
##  munsell        0.5.0      2018-06-12 [1] CRAN (R 3.5.0)                       
##  mvtnorm        1.0-8      2018-05-31 [1] CRAN (R 3.5.0)                       
##  nlme           3.1-137    2018-04-07 [1] CRAN (R 3.5.1)                       
##  pcaPP          1.9-73     2018-01-14 [1] CRAN (R 3.5.0)                       
##  pillar         1.3.1      2018-12-15 [1] CRAN (R 3.5.0)                       
##  pkgbuild       1.0.2      2018-10-16 [1] CRAN (R 3.5.0)                       
##  pkgconfig      2.0.2      2018-08-16 [1] CRAN (R 3.5.0)                       
##  pkgdown        1.3.0.9000 2019-02-06 [1] Github (r-lib/pkgdown@7e788c4)       
##  pkgload        1.0.2      2018-10-29 [1] CRAN (R 3.5.0)                       
##  plyr           1.8.4      2016-06-08 [1] CRAN (R 3.5.0)                       
##  prettyunits    1.0.2      2015-07-13 [1] CRAN (R 3.5.0)                       
##  processx       3.2.1      2018-12-05 [1] CRAN (R 3.5.0)                       
##  ps             1.3.0      2018-12-21 [1] CRAN (R 3.5.0)                       
##  purrr          0.3.0      2019-01-27 [1] CRAN (R 3.5.2)                       
##  R6             2.3.0      2018-10-04 [1] CRAN (R 3.5.0)                       
##  Rcpp           1.0.0      2018-11-07 [1] CRAN (R 3.5.0)                       
##  remotes        2.0.2      2018-10-30 [1] CRAN (R 3.5.0)                       
##  reshape        0.8.8      2018-10-23 [1] CRAN (R 3.5.0)                       
##  rlang          0.3.1      2019-01-08 [1] CRAN (R 3.5.2)                       
##  rmarkdown      1.11       2018-12-08 [1] CRAN (R 3.5.0)                       
##  robustbase     0.93-3     2018-09-21 [1] CRAN (R 3.5.0)                       
##  roxygen2       6.1.1      2018-11-07 [1] CRAN (R 3.5.0)                       
##  rprojroot      1.3-2      2018-01-03 [1] CRAN (R 3.5.0)                       
##  rrcov          1.4-7      2018-11-15 [1] CRAN (R 3.5.0)                       
##  rstudioapi     0.9.0      2019-01-09 [1] CRAN (R 3.5.2)                       
##  scales         1.0.0      2018-08-09 [1] CRAN (R 3.5.0)                       
##  sessioninfo    1.1.1      2018-11-05 [1] CRAN (R 3.5.0)                       
##  simGWAS      * 0.2.0-1    2018-11-13 [1] Github (chr1swallace/simGWAS@7ed37da)
##  snpStats       1.30.0     2018-05-01 [1] Bioconductor                         
##  stringi        1.2.4      2018-07-20 [1] CRAN (R 3.5.0)                       
##  stringr        1.3.1      2018-05-10 [1] CRAN (R 3.5.0)                       
##  survival       2.42-3     2018-04-16 [1] CRAN (R 3.5.1)                       
##  testthat       2.0.1      2018-10-13 [1] CRAN (R 3.5.0)                       
##  tibble         2.0.1      2019-01-12 [1] CRAN (R 3.5.2)                       
##  tidyselect     0.2.5      2018-10-11 [1] CRAN (R 3.5.0)                       
##  usethis        1.4.0      2018-08-14 [1] CRAN (R 3.5.0)                       
##  withr          2.1.2      2018-03-15 [1] CRAN (R 3.5.0)                       
##  xfun           0.4        2018-10-23 [1] CRAN (R 3.5.0)                       
##  xml2           1.2.0      2018-01-24 [1] CRAN (R 3.5.0)                       
##  yaml           2.2.0      2018-07-25 [1] CRAN (R 3.5.0)                       
##  zlibbioc       1.28.0     2018-10-30 [1] Bioconductor                         
## 
## [1] /Library/Frameworks/R.framework/Versions/3.5/Resources/library</code></pre>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>Researchers commonly use the size of the credible set as an estimate of the coverage probabiltiy of the causal variant in the set. It is believed to be a slightly better indicator of coverage than the threshold (obviously claimed coverage &gt; threshold).<a href="#fnref1">↩</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#simulate-gwas-summary-statistics">Simulate GWAS summary statistics</a></li>
      <li><a href="#obtain-credible-set">Obtain credible set</a></li>
      <li><a href="#obtain-corrected-coverage-estimate">Obtain corrected coverage estimate</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Anna Hutchinson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
